File: cognipy/cognipytest/cognipyunittests/InputOutputTests.java
package cognipyunittests;

import CogniPy.*;
import NUnit.Framework.*;
import java.util.*;
import java.io.*;

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [TestFixture] public class InputOutputTests
public class InputOutputTests
{
	public static String getAssemblyDirectory()
	{
		String codeBase = Assembly.GetExecutingAssembly().CodeBase;
		UriBuilder uri = new UriBuilder(codeBase);
		String path = Uri.UnescapeDataString(uri.Path);
		return (new File(path)).getParent();
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [Test] public void LoadFileWithReference()
	public final void LoadFileWithReference()
	{
		CogniPySvr feClient = new CogniPySvr();

		feClient.LoadCnl(Path.Combine(getAssemblyDirectory(), "TestFiles", "CSHC.encnl"), true, true, false);
		String[] instances = new String[] {"Dynamic[sfo]", "Answer-2-1-10-1"};
		java.lang.Iterable<CogniPy.models.InstanceDescription> result = feClient.DescribeInstancesByName(instances);
		CollectionAssert.AreEquivalent(instances, result.Select(x -> x.Instance).ToArray());
	}


//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [Test] public void ToCnlList()
	public final void ToCnlList()
	{
		ArrayList<String> cnlSentences = new ArrayList<String>(Arrays.asList("John is a man.", "Every man is a human-being."));
		CogniPySvr feClient = new CogniPySvr();
		feClient.LoadCnlFromString(tangible.StringHelper.join("\r\n", cnlSentences), true, false, false);
		ArrayList<String> cnlOut = feClient.ToCNLList(true, true, true);

		assert cnlSentences.size() == cnlOut.size();
		for (int i = 0; i < cnlOut.size(); i++)
		{
			assert cnlSentences.get(i) == cnlOut[i];
		}
	}

	private void CheckStatementCount(CogniPyStatement stmt, int Nconcept, int Ninstances, int NRoles, int NDataRole)
	{
		assert stmt.Concepts.size() == Nconcept;
		assert stmt.Instances.size() == Ninstances;
		assert stmt.Roles.size() == NRoles;
		assert stmt.DataRoles.size() == NRoles;
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [Test] public void ToCnlStatementList()
	public final void ToCnlStatementList()
	{
		ArrayList<String> cnlSentences = new ArrayList<String>(Arrays.asList("John is a man.", "Every man is a human-being."));
		CogniPySvr feClient = new CogniPySvr();
		feClient.LoadCnlFromString(tangible.StringHelper.join("\r\n", cnlSentences), true, false, false);
		ArrayList<CogniPy.CogniPyStatement> cnlOut = feClient.ToCNLStatementList(true);

		assert cnlSentences.size() == cnlOut.size();
		for (int i = 0; i < cnlOut.size(); i++)
		{
			assert cnlOut[i].CnlStatement == cnlSentences.get(i);
			if (i == 0)
			{
				CheckStatementCount(cnlOut[i], 1, 1, 0, 0);
			}
			else if (i == 1)
			{
				CheckStatementCount(cnlOut[i], 2, 0, 0, 0);
			}
		}
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//ORIGINAL LINE: [Test] public void RuleDebuggerTest()
	public final void RuleDebuggerTest()
	{
		ArrayList<String> cnlSentences = new ArrayList<String>(Arrays.asList("John is a man.", "Mary is a woman.", "John has-friend Mary.", "Mary has-name equal-to 'Mary'.", "If a man has-friend a woman and the woman has-name equal-to 'Mary' then the man is an happy-man."));
		CogniPySvr feClient = new CogniPySvr();
		String ruleId = feClient.GetStatementId(cnlSentences.get(4));
		feClient.SetDebugListener((statementId, elements) ->
		{
				assert ruleId == statementId;
				assert 2 == elements.Count();
				assert "John" == elements[0].Value.toString();
				assert "man" == elements[0].Name;
				assert "Mary" == elements[1].Value.toString();
				assert "woman" == elements[1].Name;
		}, (s, c) -> Tuple.Create(s, c));
		feClient.LoadCnlFromString(tangible.StringHelper.join("\r\n", cnlSentences), true, true, false);

	}
}

File: cognipy/CogniPyUnitTests/InputOutputTests.cs
using CogniPy;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Reflection;

namespace CogniPyUnitTests
{
    [TestFixture]
    public class InputOutputTests
    {
        public static string AssemblyDirectory
        {
            get
            {
                string codeBase = Assembly.GetExecutingAssembly().CodeBase;
                UriBuilder uri = new UriBuilder(codeBase);
                string path = Uri.UnescapeDataString(uri.Path);
                return Path.GetDirectoryName(path);
            }
        }

        [Test]
        public void LoadFileWithReference()
        {
            var feClient = new CogniPySvr();

            feClient.LoadCnl(Path.Combine(AssemblyDirectory, "TestFiles", "CSHC.encnl"), true, true);
            var instances = new string[] { "Dynamic[sfo]", "Answer-2-1-10-1" };
            var result = feClient.DescribeInstancesByName(instances);
            CollectionAssert.AreEquivalent(instances, result.Select(x => x.Instance).ToArray());
        }


        [Test]
        public void ToCnlList()
        {
            var cnlSentences = new List<string>() {
                    "John is a man.",
                    "Every man is a human-being."
            };
            var feClient = new CogniPySvr();
            feClient.LoadCnlFromString(string.Join("\r\n", cnlSentences), true, false);
            var cnlOut = feClient.ToCNLList(true, true, true);

            Assert.AreEqual(cnlSentences.Count(), cnlOut.Count());
            for (int i = 0; i < cnlOut.Count(); i++)
                Assert.AreEqual(cnlSentences[i], cnlOut[i]);
        }

        void CheckStatementCount(CogniPyStatement stmt, int Nconcept, int Ninstances, int NRoles, int NDataRole)
        {
            Assert.AreEqual(stmt.Concepts.Count(), Nconcept);
            Assert.AreEqual(stmt.Instances.Count(), Ninstances);
            Assert.AreEqual(stmt.Roles.Count(), NRoles);
            Assert.AreEqual(stmt.DataRoles.Count(), NRoles);
        }

        [Test]
        public void ToCnlStatementList()
        {
            var cnlSentences = new List<string>() {
                    "John is a man.",
                    "Every man is a human-being."
            };
            var feClient = new CogniPySvr();
            feClient.LoadCnlFromString(string.Join("\r\n", cnlSentences), true, false);
            var cnlOut = feClient.ToCNLStatementList(true);

            Assert.AreEqual(cnlSentences.Count(), cnlOut.Count());
            for (int i = 0; i < cnlOut.Count(); i++)
            {
                Assert.AreEqual(cnlOut[i].CnlStatement, cnlSentences[i]);
                if (i == 0)
                    CheckStatementCount(cnlOut[i], 1, 1, 0, 0);
                else if (i == 1)
                    CheckStatementCount(cnlOut[i], 2, 0, 0, 0);
            }
        }

        [Test]
        public void RuleDebuggerTest()
        {
            var cnlSentences = new List<string>() {
                    "John is a man.",
                    "Mary is a woman.",
                    "John has-friend Mary.",
                    "Mary has-name equal-to 'Mary'.",
                    "If a man has-friend a woman and the woman has-name equal-to 'Mary' then the man is an happy-man."
            };
            var feClient = new CogniPySvr();
            var ruleId = feClient.GetStatementId(cnlSentences[4]);
            feClient.SetDebugListener((statementId, elements) =>
            {
                Assert.AreEqual(ruleId, statementId);
                Assert.AreEqual(2, elements.Count());
                Assert.AreEqual("John", elements[0].Value.ToString());
                Assert.AreEqual("man", elements[0].Name);
                Assert.AreEqual("Mary", elements[1].Value.ToString());
                Assert.AreEqual("woman", elements[1].Name);
            }, (s, c) => Tuple.Create(s, c));
            feClient.LoadCnlFromString(string.Join("\r\n", cnlSentences), true, true);

        }
        
        [Test]
        public void LoadRdfOntologies()
        {
            var feClient = new CogniPySvr();

            Assert.DoesNotThrow(() => feClient.LoadRdf(Path.Combine(AssemblyDirectory, "TestFiles", "AEO.owl"), true, true,true));
            Assert.DoesNotThrow(() => feClient.LoadRdf(Path.Combine(AssemblyDirectory, "TestFiles", "BAMS.owl"), true, true,true));
        }
    }
}
